react-app/src/components/PortfolioScreen/PortfolioScreenAPI.ts

```diff
@@ -104,18 +104,21 @@ export const usePortfolioQuery = (): {
         (delegation) => delegation.delegation.validatorAddress
       );
 
-      const rewards = await distribution.getDelegationRewardsByValidators(
-        address,
-        validatorAddresses
-      );
+      const [rewards, validators, expectedReturns] = await Promise.all([
+        distribution.getDelegationRewardsByValidators(
+          address,
+          validatorAddresses
+        ),
+        stakingAPI.getValidators(validatorAddresses),
+        distribution.getBatchValidatorExpectedReturn(validatorAddresses),
```

Comment:
`getValidatorExpectedReturn` also gets validator? is there more efficient way to handle this... so many nested promise all :sosad:

Reply:
I suppose can return expected returns in validators, but since we already used useCallback there won't be performance issues?

Reply:
oh i was talking about both `stakingAPI.getValidators` and `distribution.getBatchValidatorExpectedReturn` gets validator from rpc